cmd/geoipupdate/main.go

```diff
@@ -274,7 +274,7 @@ func maybeUpdate(
 	if err != nil {
 		return errors.Wrap(err, "error creating request")
 	}
-	if !strings.HasPrefix(editionID, "GeoLite2") {
+	if config.AccountID != 0 {
```

Comment:
It isn't a big deal, but if you set account ID to `0` with a different license key, that license key would be ignored. In the C version, we error out in this situation when parsing the config.

Reply:
I tried to fix this with the following code and got test failures:

```diff
diff --git a/cmd/geoipupdate/config.go b/cmd/geoipupdate/config.go
index 2d6b9c7..98dc7a0 100644
--- a/cmd/geoipupdate/config.go
+++ b/cmd/geoipupdate/config.go
@@ -147,6 +147,10 @@ func NewConfig(
 	}
 	config.Proxy = proxyURL
 
+	if config.AccountID == 0 && config.LicenseKey != "000000000000" {
+		return nil, errors.New("setting an `AccountID` option of 0 with a `LicenseKey` option other than 000000000000 is disallowed")
+	}
+
 	// We used to recommend using 999999 / 000000000000 for free downloads and
 	// many people still use this combination. We need to check for the
 	// 000000000000 license key to ensure that a real AccountID of 999999 will
```

```
nchelluri@grugbarn:~/dev/geoipupdate2/cmd/geoipupdate [nchelluri/tweak-free-download-check *>] $ go test
--- FAIL: TestNewConfig (0.00s)
	config_test.go:347: 
			Error Trace:	config_test.go:347
			Error:      	Received unexpected error:
			            	setting an `AccountID` option of 0 with a `LicenseKey` option other than 000000000000 is disallowed
			            	_/home/nchelluri/dev/geoipupdate2/cmd/geoipupdate.NewConfig
			            		/home/nchelluri/dev/geoipupdate2/cmd/geoipupdate/config.go:151
			            	_/home/nchelluri/dev/geoipupdate2/cmd/geoipupdate.TestNewConfig
			            		/home/nchelluri/dev/geoipupdate2/cmd/geoipupdate/config_test.go:345
			            	testing.tRunner
			            		/usr/local/go/src/testing/testing.go:746
			            	runtime.goexit
			            		/usr/local/go/src/runtime/asm_amd64.s:2337
			Test:       	TestNewConfig
			Messages:   	Deprecated options
	config_test.go:351: 
			Error Trace:	config_test.go:351
			Error:      	Not equal: 
			            	expected: &main.Config{AccountID:0, DatabaseDirectory:"/tmp", EditionIDs:[]string{"GeoIP2-City"}, LicenseKey:"abcd", LockFile:"/tmp/.geoipupdate.lock", PreserveFileTimes:false, Proxy:(*url.URL)(nil), URL:"https://updates.maxmind.com"}
			            	actual  : (*main.Config)(nil)
			            	
			            	Diff:
			            	--- Expected
			            	+++ Actual
			            	@@ -1,13 +1,2 @@
			            	-(*main.Config)({
			            	- AccountID: (int) 0,
			            	- DatabaseDirectory: (string) (len=4) "/tmp",
			            	- EditionIDs: ([]string) (len=1) {
			            	-  (string) (len=11) "GeoIP2-City"
			            	- },
			            	- LicenseKey: (string) (len=4) "abcd",
			            	- LockFile: (string) (len=22) "/tmp/.geoipupdate.lock",
			            	- PreserveFileTimes: (bool) false,
			            	- Proxy: (*url.URL)(<nil>),
			            	- URL: (string) (len=27) "https://updates.maxmind.com"
			            	-})
			            	+(*main.Config)(<nil>)
			            	 
			Test:       	TestNewConfig
			Messages:   	Deprecated options
	config_test.go:347: 
			Error Trace:	config_test.go:347
			Error:      	Received unexpected error:
			            	setting an `AccountID` option of 0 with a `LicenseKey` option other than 000000000000 is disallowed
			            	_/home/nchelluri/dev/geoipupdate2/cmd/geoipupdate.NewConfig
			            		/home/nchelluri/dev/geoipupdate2/cmd/geoipupdate/config.go:151
			            	_/home/nchelluri/dev/geoipupdate2/cmd/geoipupdate.TestNewConfig
			            		/home/nchelluri/dev/geoipupdate2/cmd/geoipupdate/config_test.go:345
			            	testing.tRunner
			            		/usr/local/go/src/testing/testing.go:746
			            	runtime.goexit
			            		/usr/local/go/src/runtime/asm_amd64.s:2337
			Test:       	TestNewConfig
			Messages:   	CRLF line ending works
	config_test.go:351: 
			Error Trace:	config_test.go:351
			Error:      	Not equal: 
			            	expected: &main.Config{AccountID:0, DatabaseDirectory:"/tmp", EditionIDs:[]string{"GeoIP2-City"}, LicenseKey:"123", LockFile:"/tmp/.geoipupdate.lock", PreserveFileTimes:false, Proxy:(*url.URL)(nil), URL:"https://updates.maxmind.com"}
			            	actual  : (*main.Config)(nil)
			            	
			            	Diff:
			            	--- Expected
			            	+++ Actual
			            	@@ -1,13 +1,2 @@
			            	-(*main.Config)({
			            	- AccountID: (int) 0,
			            	- DatabaseDirectory: (string) (len=4) "/tmp",
			            	- EditionIDs: ([]string) (len=1) {
			            	-  (string) (len=11) "GeoIP2-City"
			            	- },
			            	- LicenseKey: (string) (len=3) "123",
			            	- LockFile: (string) (len=22) "/tmp/.geoipupdate.lock",
			            	- PreserveFileTimes: (bool) false,
			            	- Proxy: (*url.URL)(<nil>),
			            	- URL: (string) (len=27) "https://updates.maxmind.com"
			            	-})
			            	+(*main.Config)(<nil>)
			            	 
			Test:       	TestNewConfig
			Messages:   	CRLF line ending works
FAIL
exit status 1
FAIL	_/home/nchelluri/dev/geoipupdate2/cmd/geoipupdate	0.097s
nchelluri@grugbarn:~/dev/geoipupdate2/cmd/geoipupdate [nchelluri/tweak-free-download-check *>] $ 
```

What would you suggest?

Reply:
Yeah, they can be updated. Looks like they are using account ID 0 as though it were a valid account ID.

Reply:
Fixed in b62e89d68550576c4c955211969b62e3d97449cf